src/ngRoute/route.js

```diff
@@ -160,7 +160,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = route;
```

Comment:
I believe that the intention of the copying the object was to avoid the possibility of the user mutating the passed in object after the router partially processed it. This could get the router into an unknown state.

By copying we know for sure that the state we read has not changed.

The problem this causes is that if the `templateUrl` is set to be a TrustedResourceUrl instance, the instance is not copied correctly and the `templateUrl` fails to be properly unwrapped by sce later.

Since both `template` and `templateUrl` properties are both (wrapped) immutable strings, we could use `shallowCopy` instead to protect the router from internal state corruption while not corrupting the sce wrappers.

Another option is to teach angular.copy how to copy sce wrappers safely, but given that there could be custom implementations of these wrappers (e.g. at google we have our own), making the the mechanism generic might be tricky.

`shallowCopy` seems to give us all of the benefits with little down-sides, so I'd prefer that one.